src/ripple/app/transactors/CancelOffer.cpp

```diff
@@ -42,50 +42,52 @@ class CancelOffer
 
     }
 
-    TER doApply () override
+    TER preCheck () override
     {
-        std::uint32_t const uOfferSequence = mTxn.getFieldU32 (sfOfferSequence);
-        std::uint32_t const uAccountSequenceNext = mTxnAccount->getFieldU32 (sfSequence);
-
-        m_journal.debug <<
-            "uAccountSequenceNext=" << uAccountSequenceNext <<
-            " uOfferSequence=" << uOfferSequence;
-
         std::uint32_t const uTxFlags (mTxn.getFlags ());
 
         if (uTxFlags & tfUniversalMask)
         {
-            m_journal.trace <<
-                "Malformed transaction: Invalid flags set.";
+            m_journal.trace << "Malformed transaction: " <<
+                "Invalid flags set.";
             return temINVALID_FLAG;
         }
 
-        if (!uOfferSequence || uAccountSequenceNext - 1 <= uOfferSequence)
+        std::uint32_t const uOfferSequence = mTxn.getFieldU32 (sfOfferSequence);
```

Comment:
Actually, this can't be used in that way here. `sfOfferSequence` is a required field for `OfferCancel` - what we're checking is whether the specified sequence number is valid.